opm/parser/eclipse/Deck/DeckView.hpp

```diff
@@ -0,0 +1,83 @@
+/*
+  Copyright 2021 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef DECKVIEW_HPP
+#define DECKVIEW_HPP
+
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+
+namespace Opm {
+
+
+class DeckView {
+public:
+    typedef std::vector<std::reference_wrapper<const DeckKeyword>> storage_type;
+
+
+    struct Iterator : public storage_type::iterator {
+        Iterator(storage_type::const_iterator inner_iter) :
+            inner(inner_iter)
+        {}
+
+        const DeckKeyword& operator*()  { return this->inner->get(); }
+        const DeckKeyword* operator->() { return &this->inner->get(); }
+
+        Iterator& operator++()    { this->inner++; return *this; }
```

Comment:
Ehhh - here I did not really understand what you mean; as in:
```
Iterator& operator++(int){ auto tmp_inner = this->inner; ++this->inner; return Iterator(tmp_inner); }
```

Reply:
>> You should probably implement the preincrement operator in terms of preincrement of the contained storage iterator.
>
> here I did not really understand what you mean;

To increment `this->inner` using `++this->inner` instead of `this->inner++`, in other words to write it as
```C++
Iterator& operator++() { ++this->inner; return *this; }
```

Reply:
OK - I understand - for performance?

Reply:
> for performance?

Yes.  It's the general difference between `++i` and `i++`.  The former returns the new value while the latter returns the old value.  This is also the reason why C programmers occasionally say
```C++
C++; /* Make C bigger--return old value */
```
I may or may not have said so in the past myself.